src/Microsoft.AspNetCore.SignalR.Redis/RedisHubLifetimeManager.cs

```diff
@@ -187,7 +188,7 @@ public override Task OnConnectedAsync(HubConnectionContext connection)
 
             if (connection.User.Identity.IsAuthenticated)
             {
-                var userChannel = _channelNamePrefix + ".user." + connection.User.Identity.Name;
+                var userChannel = _channelNamePrefix + ".user." + connection.User.FindFirst(ClaimTypes.NameIdentifier)?.Value;
```

Comment:
Is it a valid case that user is Authenticated but don't have `NameIdentifier` claim?

> Also, you need to have this below where we publish to the user channel
Sorry, not sure I get that correctly. You mean `InvokeUserAsync` method?